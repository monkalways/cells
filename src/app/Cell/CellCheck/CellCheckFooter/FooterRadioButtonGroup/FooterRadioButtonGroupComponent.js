import React from 'react';
import PropTypes from 'prop-types';
import {
  FormControlLabel,
  Radio,
  RadioGroup,
  Typography,
  withStyles,
} from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  radioButtonValue: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  onRadioGroupChange: PropTypes.func.isRequired,
};

const FooterRadioButtonGroup = ({
  classes,
  radioButtonValue,
  disabled,
  onRadioGroupChange,
}) => (
  <RadioGroup
    name="cellCheckRadio"
    className={classes.group}
    value={radioButtonValue}
    onChange={onRadioGroupChange}
  >
    <FormControlLabel
      value="visual"
      control={<Radio className={classes.radioButton} />}
      label={<Typography variant="body1">Visual - All</Typography>}
      disabled={disabled}
    />
    <FormControlLabel
      value="verbal"
      control={<Radio className={classes.radioButton} />}
      label={<Typography variant="body1">Verbal - All</Typography>}
      disabled={disabled}
    />
  </RadioGroup>
);

FooterRadioButtonGroup.propTypes = propTypes;

export default withStyles((theme) => ({
  group: {
    marginTop: theme.spacing.unit * 0.7,
    marginLeft: -theme.spacing.unit * 15,
  },
  radioButton: {
    padding: theme.spacing.unit,
  },
}))(FooterRadioButtonGroup);
