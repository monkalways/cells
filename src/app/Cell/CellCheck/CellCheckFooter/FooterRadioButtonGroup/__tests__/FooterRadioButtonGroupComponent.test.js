import React from 'react';
import { shallow } from 'enzyme';

import { RadioGroup } from '@material-ui/core';
import { FooterRadioButtonGroupComponent } from '../FooterRadioButtonGroupComponent';

describe('FooterRadioButtonGroupComponent', () => {
  let onRadioGroupChange;

  beforeAll(() => {
    onRadioGroupChange = jest.fn();
  });

  const setup = () => {
    const classes = { group: '', radioButton: '' };
    const radioButtonValue = 'visual';
    const disable = false;
    return shallow(<FooterRadioButtonGroupComponent
      classes={classes}
      radioButtonValue={radioButtonValue}
      disable={disable}
      onRadioGroupChange={onRadioGroupChange}
    />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    wrapper.find(RadioGroup).simulate('change');
    expect(onRadioGroupChange).toBeCalled();
  });
});
