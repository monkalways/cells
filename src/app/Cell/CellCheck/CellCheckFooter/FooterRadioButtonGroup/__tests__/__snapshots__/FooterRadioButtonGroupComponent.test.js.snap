// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterRadioButtonGroupComponent should render 1`] = `
<RadioGroup
  className=""
  name="cellCheckRadio"
  onChange={[MockFunction]}
  value="visual"
>
  <WithStyles(FormControlLabel)
    control={
      <WithStyles(Radio)
        className=""
      />
    }
    label={
      <WithStyles(Typography)
        variant="body1"
      >
        Visual - All
      </WithStyles(Typography)>
    }
    value="visual"
  />
  <WithStyles(FormControlLabel)
    control={
      <WithStyles(Radio)
        className=""
      />
    }
    label={
      <WithStyles(Typography)
        variant="body1"
      >
        Verbal - All
      </WithStyles(Typography)>
    }
    value="verbal"
  />
</RadioGroup>
`;
